### PR TITLE
feat: <Badge> update, add <PlanIndicator>s

### PR DESCRIPTION
**Pull request by fk**
This PR updates the `Badge` component

- to support
  - `tone` (the existing `SUCCESS`, `WARNING`, `BRAND`, `DANGER`, `NEUTRAL`)
  - `size` (`S | M`)
  - `textVariant` (`DEFAULT | CAPS`)
- to accept an `Icon`, and to define its position and appearance
- so that the existing variant `PILL` now matches what we call `SOLID` for `Notification`, while the existing `STATUS` variant is the "DEFAULT", less emphasized state of the component

It also extends the component Sandbox with the above options, and links the Figma design.

image. 
https://deploy-preview-293--gatsby-interface.netlify.app/?path=/story/badge--sandbox

This PR also adds four `PlanIndicator` icons (based on `<Badge size="M">`), courtesy of @Elanhant <3:

image. 
https://deploy-preview-293--gatsby-interface.netlify.app/?path=/story/planindicator--plan-types

/cc @nachiketkumar

### CODE REVIEW

File: __tests__/icons.test.js [hunk 1]
```diff
@@ -1,6 +1,5 @@
 import React from "react"
-import { render } from "@testing-library/react"
-
+import { renderWithTheme } from "../src/utils/testing"
```
Comment (fk) [inline __tests__/icons.test.js:2]: 👨‍🎓🙏

**Comment by fk** [comment]
Should we rename `PILL` to `SOLID`, and `STATUS` to `DEFAULT`?

**Comment by mfrachet** [comment]
I'm merging this to have quick feedback on .com. If something goes wrong, we can reiterate and adjust 💪

**Comment by fk** [comment]
💥 💥 💥

**Pull request merged**